src/SFA.DAS.EmployerAccounts.Events/Messages/Message.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace SFA.DAS.EmployerAccounts.Events.Messages
+{
+    public abstract class Message
+    {
+        protected Message(string signedByName=null, long accountId=0)
+        {
+            SignedByName = signedByName;
```

Comment:
SignedByName : this looks like its the external user id that created the message so it should be renamed to reflect this. I'd take it out of the generic message base as well and create a new message for MA that can be used when the message is created by a user as not all maybe.

Reply:
It's not exteralUserId, it's the name of the person who pressed the button

Reply:
externalUserId is the UserId of the person who's logged in and is used to get the name. If you need to persist the name then it may be useful to store it in the message however if we are processing these quickly that may not be an issue. The reason I was saying to would be better to store the ID is that you then can get further details like email address if you wanted without having to add any more code to the message. I think signedBy sounds wrong as well as it really the person who created it.

Reply:
Just to add one more thing. When we designed the messaging system we didn't take into account the need for data to be persisted in the messages themselves. So sending the userId would then be picked up by activities and the user's name would then be persisted in the activity record itself. We could change this of course but that's why I mentioned it as the messages were not designed with this in mind. Events were designed like this which is why they cause some concern (especially with replaying external data sources).

Reply:
This could be the reason why it looks like i'm taking a different view as I'm seeing messages a short term signals that alert listeners and then those listeners collect the information they need to the IDs stored in the message. Its similar to the way Michael did the API calls for the RDS project.